ratis-server/src/main/java/org/apache/ratis/server/impl/ServerState.java

```diff
@@ -221,10 +221,10 @@ boolean hasLeader() {
   /**
    * Become a candidate and start leader election
    */
-  long initElection() {
+  long initElection(boolean preVote) {
```

Comment:
If `preVote` is true, you will not change `currentTerm`, meanwhile change `voteFor`, which may violate the raft algorithm that one can only vote once in a specific term.

Reply:
@GlenGeng Thanks for review, I define askForPreVotes method to process pre vote, in this method, it will not change voteFor.